frontend/src/components/common/header/components/search-bar/search-bar.tsx

```diff
@@ -0,0 +1,31 @@
+import searchIcon from 'assets/img/search-icon.svg';
+import { FC } from 'common/types/types';
+import { useState } from 'hooks/hooks';
+import { coursesApi } from 'services/services';
+
+import { SearchIcon } from './components/search-icon/search-icon';
+import styles from './styles.module.scss';
+
+const SearchBar: FC = () => {
+  const [searchValue, setSearchValue] = useState('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    setSearchValue(event.target.value);
+    coursesApi.getByName(event.target.value);
+  };
```

Comment:
I think implementing `debounce` function would be great here

Reply:
We already have this function in shared